digitalocean/resource_digitalocean_droplet.go

```diff
@@ -626,20 +626,38 @@ func resourceDigitalOceanDropletDelete(d *schema.ResourceData, meta interface{})
 	log.Printf("[INFO] Deleting droplet: %s", d.Id())
 
 	// Destroy the droplet
-	_, err = client.Droplets.Delete(context.Background(), id)
+	resp, err := client.Droplets.Delete(context.Background(), id)
 
-	// Handle remotely destroyed droplets
-	if err != nil && strings.Contains(err.Error(), "404 Not Found") {
+	// Handle already destroyed droplets
+	if err != nil && resp.StatusCode == 404 {
 		return nil
 	}
 
-	if err != nil {
+	_, err = waitForDropletDestroy(d, meta)
+	if err != nil && strings.Contains(err.Error(), "404") {
+		return nil
+	} else if err != nil {
 		return fmt.Errorf("Error deleting droplet: %s", err)
 	}
 
 	return nil
 }
 
+func waitForDropletDestroy(d *schema.ResourceData, meta interface{}) (interface{}, error) {
+	log.Printf("[INFO] Waiting for droplet (%s) to be destroyed", d.Id())
+
+	stateConf := &resource.StateChangeConf{
+		Pending:    []string{"active"},
+		Target:     []string{"archived"},
```

Comment:
Where do these states come from?

Reply:
`active` is the `status` attribute for a running Droplet. Realizing now that `off` should also be included as you can destroy a powered off Droplet. `archived` is an internal status for deleted Droplets. In practice, this will not be seen on the public API as it will 404 when it is destroyed.

This function will always return an error. 404 when the Droplet is successfully destroyed or a timeout if the destroy request fails unexpectedly on the DigitalOcean. I'm really just taking advantage of the Terraform's existing polling mechanism.